conda/core/solve.py

```diff
@@ -626,6 +626,8 @@ def _add_specs(self, ssc):
                     python_spec = MatchSpec(python_spec, version=pinned_version)
 
                 spec_set = (python_spec, ) + tuple(self.specs_to_add)
+                if len(conflict_specs) > 0 and self._repodata_fn == "current_repodata.json":
+                    return ssc
                 if ssc.r.get_conflicting_specs(spec_set):
```

Comment:
I'm not totally sure here, but we need to think about how the program flows from here. Do we need to raise an exception to jump out to the outer loop in install.py? Exception handling for flow control is a coffee smell. Is there a better way for us to get back to that outer loop?

Reply:
Ya, that makes sense. How about only running `find_conflicts` if using `repodata.json` and then if any conflicting specs are found using non `repodata.json` then raising an `UnsatsifiableError`. Thats what `find_conflicts` will do so the parent function already knows how to deal with that case